packages/ember-glimmer/tests/integration/syntax/each-in-test.js

```diff
@@ -305,3 +310,34 @@ moduleFor('Syntax test: {{#each-in}} undefined path', class extends RenderingTes
     this.assertText('');
   }
 });
+
+moduleFor('Syntax test: {{#each-in}}  for array', class extends RenderingTest {
+  ['@test it iterate over array with `in` instead of walking over elements'](assert) {
+    let arr = [1, 2, 3];
+
+    this.render(strip`
+      {{#each-in arr as |key value|}}
+        [{{key}}:{{value}}]
+      {{/each-in}}`, { arr });
+
+    this.assertText('[0:1][1:2][2:3]');
+
+    this.runTask(() => {
+      set(arr, 'someKey', 'someKeyValue');
+      this.rerender();
+    });
+
+    this.assertText('[0:1][1:2][2:3][someKey:someKeyValue]');
+
+    this.runTask(() => {
+      delete arr.someKey;
+      this.rerender();
+    });
+
+    this.assertText('[0:1][1:2][2:3]');
+
+    this.runTask(() => set(this.context, 'arr', []));
+
+    this.assertText('');
```

Comment:
Missing the "R" step after this:

`this.runTask(() => set(this.context, 'arr', [1, 2, 3]));`

(which reset the text output to the original value, except using a fresh array)